journal/202307.md

```diff
@@ -0,0 +1,596 @@
+- Improved and polished [wallet tests](https://github.com/decred/dcrwallet/pull/2260), including removing redundant code.
+
+
+### dcrctl
+
+_[dcrctl](https://github.com/decred/dcrctl) is a command-line client for dcrd and dcrwallet._
+
+The following work has been merged in `master` towards future releases:
+
+- [Removed stale dependencies](https://github.com/decred/dcrctl/pull/64) from the main module, and updated the [dcrd and dcrwallet modules](https://github.com/decred/dcrctl/pull/67) to their latest versions. Notably, this makes dcrd's new verbose results of `getblock` and `getblockheader` available.
+- Updated the README to recommend that developers use local [Go Workspaces](https://github.com/decred/dcrctl/pull/66) when working with development versions of dcrd and dcrwallet.
+
+
+### Decrediton
+
+_[Decrediton](https://github.com/decred/decrediton) is a full-featured desktop wallet app with integrated voting, StakeShuffle mixing, Lightning Network, DEX trading, and more. It runs with or without a full blockchain (SPV mode)._
+
+In July, @norwnd proposed a [2FA storage for Decrediton](https://gist.github.com/norwnd/890ad642985f4e9e9f7b1dd243b21f9e) based on simple 2-of-2 multisig. The idea is to provide a more secure solution for storing DCR by using a second device to sign transactions, initially an Android smartphone. This should protect from threats like a weak wallet encryption password or a stolen laptop. As a bonus it could make multisig features more accessible to regular users who don't have skills to program a multisig solution themselves. It is currently in early discussion and feedback phase but some effort has been made to test multisig features in dcrwallet and test how much data can be passed in QR codes.
+
+
+### vspd
+
+_[vspd](https://github.com/decred/vspd) is server software used by Voting Service Providers. A VSP votes on behalf of its users 24/7 and cannot steal funds._
+
+Changes that were included in the [v1.2.1 release](https://github.com/decred/vspd/releases/tag/release-v1.2.1):
+
+- Fixed [transaction broadcasting](https://github.com/decred/vspd/pull/398) logic to not generate an error when the transaction already exists (which is not a problem). This should fix [VSP fee payment issues](https://matrix.to/#/!zefvTnlxYHPKvJMThI:decred.org/$mn3UvlfCEWxDVqxqL07-K6bnPl2HKN-hPZwUL4n4DSg) in some cases involving chains of multiple unconfirmed transactions.
```

Comment:
This missing error condition was noticed when investigating the broken retry issue in dcrwallet. It's a preventative fix, I don't think it was causing any issues in the real world. In the unlikely case that this were to occur in production, it could have resulted in a client wallet paying two fees for a single ticket.

Reply:
Reworded and clarified it was a reported issue in the wild. Thanks.